demos/pinter/src/main.rs

```diff
@@ -0,0 +1,116 @@
+extern crate image;
+extern crate dungenon;
+extern crate cast;
+extern crate rand;
+
+use std::io;
+use std::path::PathBuf;
+
+use dungenon::level::GridLevel as Level;
+use dungenon::tile::{Faction};
+use dungenon::generator::{FactionGen};
+use dungenon::util::Error;
+
+use rand::os::OsRng;
+use rand::XorShiftRng;
+use rand::Rand;
+use rand::distributions::{IndependentSample, Range};
+
+use image::RgbImage;
+use image::Rgb;
+
+use std::collections::HashMap;
+
+fn main() {
+    let mut colors = HashMap::new();
+    let mut index = 0;
+    for r in 0..256 {
+        for g in 0..256 {
+            for b in 0..256 {
+                colors.insert(Faction::Faction(index), Rgb([r as u8, g as u8, b as u8]));
+                index += 1;
+            }
+        }
+    }
+    println!("Initializing level...");
+    println!("Input level width: ");
+    let x = usize_from_cmd();
+
+
+    println!("Input level height: ");
+    let y = usize_from_cmd();
+
+    let mut level = Level::new_filled_with(Faction::Neutral, x, y);
+    populate_level(&mut level, index);
+    let mut factiongen = FactionGen::new();
+    println!("How many iterations");
+    let iterations = u64_from_cmd();
+    let mut buffer = level.clone();
+    for i in 0..iterations {
+        factiongen.generate(&mut level, &mut buffer);
+        println!("{} iterations done.", i);
+    }
+    faction_png_export(String::from("picture"), &colors, &mut level);
+}
+
+fn populate_level(level: &mut Level<Faction>, rangemax: usize) {
+    let range = Range::new(0,rangemax);
+    let mut rand = XorShiftRng::rand(&mut OsRng::new().unwrap());
+    for x in 0..level.get_width() {
+        for y in 0..level.get_height() {
+            match level.get_mut_tile(x, y) {
+                Ok(t) => {
+                    *t = Faction::Faction(range.ind_sample(&mut rand));
+                },
+                _ => unreachable!("aaaa"),
+            }
+        }
+    }
+}
+
+fn faction_png_export(name: String, colors: &HashMap<Faction, Rgb<u8>>, level: &mut Level<Faction>) {
+    let mut level_image = RgbImage::new(level.get_width() as u32, level.get_height() as u32);
+    for x in 0 .. level.get_width() {
+        for y in 0 .. level.get_height() {
+            level_image.put_pixel(x as u32, y as u32, faction_to_color(colors, &level.get_tile(x,y)));
+        }
+    }
+    let mut p = PathBuf::new();
+    p.push(name);
+    p.set_extension("png");
+    level_image.save(p.as_path()).expect("Something went wrong when saving the png.");
+}
+
+fn faction_to_color(colors: &HashMap<Faction, Rgb<u8>>, tile: &Result<&Faction, Error>) -> Rgb<u8> {
+    match *tile {
+        Ok(ref tile) => {
+                match colors.get(tile) {
+                    Some(color) => color.clone(),
+                    None => Rgb([0u8, 0u8, 0u8]),
```

Comment:
There is faction of this color.

Reply:
Does it matter? The whole thing should be populated with... oh right. I can just make it panic in the none case can't I.

Reply:
There is none case right? When you start the whole level is not filled?

Reply:
Or am I mistaken?

Reply:
the whole level is filled